File: MyEclipseProject/20180718/src/com/zrf/ex/bean/Person.java

/**
* @Title: Person.java
* @Package com.zrf.ex.bean
* @Description: 构建Person类
* @author ZRF
* @date 2018年7月18日
* @version V1.0  
*/

package com.zrf.ex.bean;

import com.zrf.ex.meiju.Month;

/**
 * @ClassName: Person
 * @Description: 构建Person类
 * @author ZRF
 * @date 2018年7月18日
 */

public class Person {
	private String name;
	private double height;
	private Month month;

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public double getHeight() {
		return height;
	}

	public void setHeight(double height) {
		this.height = height;
	}

	public Month getMonth() {
		return month;
	}

	public void setMonth(Month month) {
		this.month = month;
	}

	public Person() {
		super();
	}

	public Person(String name, double height, Month month) {
		super();
		this.name = name;
		this.height = height;
		this.month = month;
	}

	@Override
	public String toString() {
		return "Person [name=" + name + ", height=" + height + ", month=" + month + "]";
	}
}

File: MyEclipseProject/20180821/src/com/zrf/ex/Person.scala
package com.zrf.ex

class Person(val name: String, var age: Int) {
  var height: Double = 170
  println("gougou")

  //辅助构造器(辅助构造器使用之前必先调用主构造器)
  def this() {
    //调用主构造器
    this("wanger", 17)
    height = 180
  }

  //重写方法
  override def toString(): String = {
    name + "\t" + age + "\t" + height
  }

}

object Person { //伴生对象
  def main(args: Array[String]): Unit = {
    go()
  }

  def go(): Unit = {
    var p = new Person("zhangsan", 18);
    println(p)

    //修改值
    p.age = 20
    p.height = 175
    println(p.name + "\t" + p.age + "\t" + p.height)

    println(new Person("lisi", 21))

    //调用辅助构造器
    var p1 = new Person
    println(p1)
  }
}